improver/utilities/mathematical_operations.py

```diff
@@ -356,3 +356,52 @@ def process(self, cube):
         integrated_cube = self.perform_integration(upper_bounds_cube, lower_bounds_cube)
 
         return integrated_cube
+
+
+def alinfit(X, Y, axis=-1, gradient_only=False):
+    """Uses a simple linear fit approach to calculate the
+    gradient. Although equivalent, this
+    approach is much faster than using scipy lstsq to fit the
+    data.
+
+    Args:
+        X:
+            ndarray, X axis data
+        Y:
+            ndarray, Y axis data
+        axis:
+            Optional argument, specifies the axis to operate on.
+        gradient_only:
+            Boolean, if true only returns the gradient.
+
+    Returns:
+        The gradient and the y-intercept in 2 1d numpy arrays.
+    """
+    if not isinstance(axis, tuple):
+        axis = (axis,)
+
+    # Check that there are no spurious NaNs in one of the arrays
+    # only - this will mess up the mean.
+    assert (np.isnan(Y) == np.isnan(X)).all
+
+    # Finds a compatible shape for the means to be reshaped into
+    assert X.shape == Y.shape
+    shape = list(X.shape)
+    for ax in axis:
+        shape[ax] = 1
+
+    X_mean = np.nanmean(X, axis=axis).reshape(shape)
+    Y_mean = np.nanmean(Y, axis=axis).reshape(shape)
+
+    X_diff = X - X_mean
+    Y_diff = Y - Y_mean
+
+    XY_cov = np.nansum(X_diff * Y_diff, axis=axis)
+    X_var = np.nansum(X_diff * X_diff, axis=axis)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable name "X_var" doesn't conform to snake_case naming style](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5319684)